Eatery/Controllers/BRB/BRBAccountManager.swift

```diff
@@ -0,0 +1,277 @@
+//
+//  AccountManager.swift
+//  Eatery
+//
+//  Created by Reade Plunkett on 10/24/19.
+//  Copyright © 2019 CUAppDev. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import WebKit
+
+struct HistoryEntry {
+
+    var description: String = ""
+    var timestamp: String = ""
+
+}
+
+enum Stage {
+
+    case loginScreen
+    case transition
+    case loginFailed
+    case finished(sessionId: String)
+
+}
+
+// MARK: - Account Settings
+
+typealias LoginInfo = (netid: String, password: String)
+
+private enum BRBAccountSettings {
+    
+    static func saveToKeychain(loginInfo: LoginInfo) {
+        let keychain = KeychainItemWrapper(identifier: "netid", accessGroup: nil)
+        keychain["netid"] = loginInfo.netid as AnyObject
+        keychain["password"] = loginInfo.password as AnyObject
+    }
+    
+    static func removeKeychainLoginInfo() {
+        let keychain = KeychainItemWrapper(identifier: "netid", accessGroup: nil)
+        keychain["netid"] = nil
+        keychain["password"] = nil
+    }
+    
+    static func loadFromKeychain() -> LoginInfo? {
+        let keychain = KeychainItemWrapper(identifier: "netid", accessGroup: nil)
+        guard let netid = keychain["netid"] as? String, let password = keychain["password"] as? String else {
+            return nil
+        }
+        return (netid: netid, password: password)
+    }
+    
+}
+
+private let loginURL = URL(string: "https://get.cbord.com/cornell/full/login.php?mobileapp=1")
+private let maxTrials = 3
+private let trialDelay = 500
+
+// MARK: - Connection Handler
+
+private protocol BRBConnectionHandlerDelegate {
+
+    func brbConnectionHandlerDelegateDidRetrieve(sessionID: String)
+    func brbConnectionHandlerDelegateDidFailLogin(with error: String)
+
+}
+
+private class BRBConnectionHandler: WKWebView, WKNavigationDelegate {
+    
+    var stage: Stage = .loginScreen
+    private var diningHistory: [HistoryEntry] = []
+    private var loginCount = 0
+    var netid: String = ""
+    var password: String = ""
+    var delegate: BRBConnectionHandlerDelegate?
+    
+    init() {
+        super.init(frame: .zero, configuration: WKWebViewConfiguration())
+        navigationDelegate = self
+        URLCache.shared.diskCapacity = 0
+        URLCache.shared.memoryCapacity = 0
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    /// Gets the HTML for the current web page and runs block after loading
+    /// HTML into a string
+    func getHTML(block: @escaping (String) -> ()){
+        evaluateJavaScript("document.documentElement.outerHTML.toString()",
+                           completionHandler: { (html: Any?, error: Error?) in
+                            if let html = html {
+                                block(html as! String)
+                            }
+        })
+    }
```

Comment:
Force unwraps are generally bad. Will the compiler take this:

    evaluateJavaScript("document.documentElement.outerHTML.toString()") { (html: Any?, error: Error?) in
        guard let html = html as? String else { return}
        block(html)
    })